layout-server/src/test/java/org/finos/vuu/layoutserver/controller/LayoutControllerTest.java

```diff
@@ -0,0 +1,170 @@
+import org.finos.vuu.layoutserver.dto.request.MetadataRequestDTO;
+import org.finos.vuu.layoutserver.dto.response.CreateLayoutResponseDTO;
+import org.finos.vuu.layoutserver.dto.response.GetLayoutResponseDTO;
+import org.finos.vuu.layoutserver.dto.response.MetadataResponseDTO;
+import org.finos.vuu.layoutserver.model.Layout;
+import org.finos.vuu.layoutserver.model.Metadata;
+import org.finos.vuu.layoutserver.service.LayoutService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.modelmapper.ModelMapper;
+
+@ExtendWith(MockitoExtension.class)
+class LayoutControllerTest {
+
+    @Mock
+    private LayoutService layoutService;
+
+    @Mock
+    private ModelMapper modelMapper;
+
+    @InjectMocks
+    private LayoutController layoutController;
+
+    private UUID validLayoutId;
+    private UUID doesNotExistLayoutId;
+    private Layout layout;
+    private Metadata metadata;
+    private LayoutRequestDTO layoutRequest;
+    private GetLayoutResponseDTO expectedLayoutResponse;
+    private List<MetadataResponseDTO> expectedMetadataResponse;
+
+    @BeforeEach
+    public void setup() {
+        validLayoutId = UUID.randomUUID();
+        doesNotExistLayoutId = UUID.randomUUID();
+        UUID metadataId = UUID.randomUUID();
+        String layoutDefinition = "Test Definition";
+
+        metadata = new Metadata();
+        metadata.setId(metadataId);
+        metadata.setName("Test Layout");
+        metadata.setUser("Test User");
+        metadata.setGroup("Test Group");
+        metadata.setScreenshot("Test Screenshot");
+
+        layout = new Layout();
+        layout.setId(validLayoutId);
+        layout.setDefinition(layoutDefinition);
+        layout.setMetadata(metadata);
+        metadata.setLayout(layout);
+
+        layoutRequest = new LayoutRequestDTO();
+        MetadataRequestDTO metadataRequestDTO = new MetadataRequestDTO();
+        metadataRequestDTO.setName(metadata.getName());
+        metadataRequestDTO.setUser(metadata.getUser());
+        metadataRequestDTO.setGroup(metadata.getGroup());
+        metadataRequestDTO.setScreenshot(metadata.getScreenshot());
+        layoutRequest.setDefinition(layout.getDefinition());
+        layoutRequest.setMetadata(metadataRequestDTO);
+
+        expectedLayoutResponse = new GetLayoutResponseDTO();
+        expectedLayoutResponse.setId(layout.getId());
+        expectedLayoutResponse.setDefinition(layout.getDefinition());
+
+        MetadataResponseDTO metadataResponse = getMetadataResponseDTO();
+        expectedLayoutResponse.setMetadata(metadataResponse);
+
+        expectedMetadataResponse = new ArrayList<>();
+        expectedMetadataResponse.add(metadataResponse);
+    }
+
+
+    @Test
+    void getLayout_validIdAndLayoutExists_returnsLayout() {
```

Comment:
What do we mean by validId here? is it just an id for which we have a layout? in that case `getLayout_layoutExists_returnsLayout()` would descriptive enough